pilot/pkg/model/context.go

```diff
@@ -884,14 +890,39 @@ func (node *Proxy) SetWorkloadLabels(env *Environment) {
 
 // DiscoverIPMode discovers the IP Versions supported by Proxy based on its IP addresses.
 func (node *Proxy) DiscoverIPMode() {
+	var wildCardIPs, localHostIPs []string
 	if networkutil.AllIPv4(node.IPAddresses) {
 		node.ipMode = IPv4
+		wildCardIPs = append(wildCardIPs, constants.WildcardAddress)
+		localHostIPs = append(localHostIPs, constants.LocalhostAddress)
 	} else if networkutil.AllIPv6(node.IPAddresses) {
 		node.ipMode = IPv6
+		wildCardIPs = append(wildCardIPs, constants.WildcardIPv6Address)
+		localHostIPs = append(localHostIPs, constants.LocalhostIPv6Address)
 	} else {
 		node.ipMode = Dual
+		wildCardIPs = append(wildCardIPs, []string{constants.WildcardAddress, constants.WildcardIPv6Address}...)
+		localHostIPs = append(localHostIPs, []string{constants.LocalhostAddress, constants.LocalhostIPv6Address}...)
 	}
 	node.GlobalUnicastIP = networkutil.GlobalUnicastIP(node.IPAddresses)
+	node.wildCardIPs = wildCardIPs
+	node.localHostIPs = localHostIPs
+}
+
+// Wildcards returns the wildCard IP addresses for current proxy
+func (node *Proxy) Wildcards() []string {
+	if len(node.wildCardIPs) == 0 {
+		node.DiscoverIPMode()
```

Comment:
Hi @hzxuzhonghu, yeah, `DiscoverIPMode` would be called one time if `len(node.wildCardIPs) == 0` or `len(node.localHosts) == 0`. 
In fact, I'm afraid that `DiscoverIPMode` is not called, so just in case. Do you suggest that remove the calling in both `Wildcards()` and `Localhosts()`?

Reply:
removed!